vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -258,21 +264,21 @@ def weight_loader(self, param: torch.nn.Parameter,
             if shard_id == 0:
                 param_data[expert_id,
                            0:shard_size, :] = loaded_weight[shard, :]
-                if is_hpu():
-                    self.hpu_static_fused_moe.w13_list[expert_id].set_weight(
+                if is_hpu() and hasattr(self.hpu_fused_moe, 'w13_list'):
```

Comment:
Maybe to make the condition more clear instead of hasattr we can check that isinstance(self.hpu_fused_moe, StaticFusedMOE)